learning_journal/tests.py

```diff
@@ -98,72 +78,195 @@ def test_home_list_returns_empty_when_empty(dummy_request):
     assert len(query_list) == 0
 
 
-def test_home_list_returns_objects_when_exist(dummy_request, add_models):
+def test_home_list_returns_objects_when_exist(dummy_request, new_session):
     """Test that the home list does return objects when the DB is populated."""
     from .views.default import home_list
+    model = Entry(title=ENTRIES[0]["title"],
+                  body=ENTRIES[0]["body"],
+                  creation_date=ENTRIES[0]["creation_date"])
+    new_session.add(model)
     result = home_list(dummy_request)
     query_list = result["posts"][:]
-    assert len(query_list) == 2
+    assert len(query_list) == 1
+
+
+def test_detail_returns_entry_1(dummy_request, new_session):
+    """Test that entry return entry one."""
+    from .views.default import detail
+    model = Entry(title=ENTRIES[0]["title"],
+                  body=ENTRIES[0]["body"],
+                  creation_date=ENTRIES[0]["creation_date"])
+    new_session.add(model)
+    dummy_request.matchdict['id'] = 1
+    result = detail(dummy_request)
+    query_reslts = result["post"]
+    assert query_reslts.title == ENTRIES[0]["title"]
+    assert query_reslts.body == ENTRIES[0]["body"]
+
+
+def test_detail_returns_entry_2(dummy_request, new_session):
+    """Test that entry return entry two."""
+    from .views.default import detail
+    model = Entry(title=ENTRIES[1]["title"],
+                  body=ENTRIES[1]["body"],
+                  creation_date=ENTRIES[1]["creation_date"])
+    new_session.add(model)
+    dummy_request.matchdict['id'] = 1
+    result = detail(dummy_request)
+    query_reslts = result["post"]
+    assert query_reslts.title == ENTRIES[1]["title"]
+    assert query_reslts.body == ENTRIES[1]["body"]
+
+
+def test_update_returns_entry_1(dummy_request, new_session):
+    """Test update returns entry two."""
+    from .views.default import update
+    model = Entry(title=ENTRIES[0]["title"],
+                  body=ENTRIES[0]["body"],
+                  creation_date=ENTRIES[0]["creation_date"])
+    new_session.add(model)
+    dummy_request.matchdict['id'] = 1
+    result = update(dummy_request)
+    query_reslts = result["post"]
+    assert query_reslts.title == ENTRIES[0]["title"]
+    assert query_reslts.body == ENTRIES[0]["body"]
+
+
+def test_update_returns_entry_2(dummy_request, new_session):
+    """Test update returns entry two."""
+    from .views.default import update
+    model = Entry(title=ENTRIES[1]["title"],
+                  body=ENTRIES[1]["body"],
+                  creation_date=ENTRIES[1]["creation_date"])
+    new_session.add(model)
+    dummy_request.matchdict['id'] = 1
+    result = update(dummy_request)
+    query_reslts = result["post"]
+    assert query_reslts.title == ENTRIES[1]["title"]
+    assert query_reslts.body == ENTRIES[1]["body"]
```

Comment:
This test does not actually test that entry 2 is returned.  There is only one entry in the table.